app/src/main/java/ch/epfl/sdp/blindly/user/UserHelper.kt

```diff
@@ -54,56 +52,65 @@ class UserHelper {
 
     }
 
-    fun signOut(activity: Activity, onComplete: OnCompleteListener<Void>) {
-        AuthUI.getInstance()
-                .signOut(activity)
-                .addOnCompleteListener(onComplete)
-    }
-
-    fun delete(activity: Activity, onComplete: OnCompleteListener<Void>) {
-        AuthUI.getInstance()
-                .delete(activity)
-                .addOnCompleteListener(onComplete)
-    }
-
-    fun isLoggedIn(): Boolean {
-        return FirebaseAuth.getInstance().currentUser != null
-    }
-
-
-    fun handleAuthResult(activity: Activity, resultCode: Int, data: Intent?): Boolean {
+    fun handleAuthResult(activity: Activity, resultCode: Int, data: Intent?): Intent? {
         val response = IdpResponse.fromResultIntent(data)
 
         if (resultCode == Activity.RESULT_OK) {
             // Successfully signed in
             val user = FirebaseAuth.getInstance().currentUser
-            return true
-            // ...
+            if (isNewUser(user!!)) {
+                return Intent(activity, ProfileHouseRules::class.java)
+            }
+            return Intent(activity, MainScreen::class.java)
         } else {
             // Sign in failed. If response is null the user canceled the
             // sign-in flow using the back button. Otherwise check
             // response.getError().getErrorCode() and handle the error.
             // ...
             if (response != null) {
                 Toast.makeText(
-                        activity.applicationContext,
-                        activity.getString(
-                                R.string.login_err,
-                                response.error?.errorCode ?: -1
-                        ),
-                        Toast.LENGTH_LONG
+                    activity.applicationContext,
+                    activity.getString(
+                        R.string.login_err,
+                        response.error?.errorCode ?: -1
+                    ),
+                    Toast.LENGTH_LONG
                 ).show()
             }
-            return false
+            return null
         }
     }
 
+    /**
+     * Check is the user has already created an account
+     *
+     * @param user: the user to test the condition on
+     * @return true if the user just created an account, false otherwise
+     */
+    private fun isNewUser(user: FirebaseUser) : Boolean {
+        val metadata = user.metadata
+        return metadata?.creationTimestamp == metadata?.lastSignInTimestamp
+    }
+
+    fun delete(activity: Activity, onComplete: OnCompleteListener<Void>) {
```

Comment:
Can't we use this function in the settings activity to delete the account ?

Reply:
I don't know, I didn't write this part but probably (though there might be a bug, which is why it's not called yet)?